rfcs/0005-music-system.md

```diff
@@ -0,0 +1,155 @@
+
+When the player enters a system with a music node defined, the following happens:
+1. All normal `playlists` are treated as invalid.
+2. The system's `track` is treated as if it were used in a single-`track` `playlist` - the only valid playlist.
+3. The new playlist is chosen, which is inevitably the system's pseudo-`playlist`. All `variant`-related playback rules are adhered to as usual.
+4. If the player stays in the system long enough for the `track` to finish, a new `playlist` is chosen - and as before, the only option is the system's music.
+5. Upon leaving the system, the `playlist` becomes invalid as any other would. At this time, the standard `playlists` would no longer be treated as invalid, and would be chosen to replace the system's music.
+
+When a mission trigger plays a `track`, the behaviour is different:
+1. The current `track` stops playback. The active `playlist` is discarded.
+2. The mission's `track` starts playing from the beginning. `Variant`-related playback rules are NOT followed.
+3. When the mission's `track` finishes, a new `playlist` is chosen as usual. There is no guarantee tha the previously playing `playlist` will be resumed here.
+
+> [!NOTE]
+> An experimental demo of this system is available at https://tibetiroka.com/music with limited functionality.
+
+# Drawbacks
+Having multiple music `layers` increases complexity and file I/O operations. It can also increase the number of audio output sinks used, though we can mix the music samples before sending them to the operating system if necessary.
+
+As music is often streamed, there is no easy way to guarantee that an audio file begins playback immediately. Therefore there is no way to keep the foreground layers in sync with the background layer completely without using cached WAV samples. (The foreground layers are meant to provide ambience, not orchestral playback, so this is probably a non-issue.)
+
+# Alternatives
+Alternatives have been discussed extensively regarding the previous music PRs. Though much of the structure is preserved from the previous solutions, there are some key differences:
+- `Variants` are now for `tracks`, not for audio files. They are also arbitrarily defined, and are not required for the combat/non-combat switching to work properly.
+  - As it isn't reasonable to expect every `track` to have a combat `variant`, the new approach enables playing dedicated combat music in scenarios where previously regular music would have been played. This is helped by moving the combat `variants` to another `playlist` from the file specification of `tracks`.
+- `Tracks` can have multiple audio layers playing concurrently.
+  - This choice was made based on the results of the creator survey, and was not known at the time of the previous discussions.
+- `Playlists` are now "play once", not "play forever". This fixes various issues where players could get stuck listening to the same few songs when many others were also available.
```

Comment:
This doesn't fully align with what I'd expect from the common definition of a playlist. Perhaps there's a more fitting term?